src/views/token-swap/components/swap-button/swap-button.view.jsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import useSwapButtonStyles from './swap-button.style'
+
+function SwapButton ({
+  quotes
+}) {
+  const classes = useSwapButtonStyles()
+  const failure = quotes.status === 'failure'
+
+  const [timeUntilValid, setTimeUntilValid] = React.useState(30000)
+
+  React.useEffect(() => {
+    setTimeUntilValid(30000 - 1000)
+    // TODO should be launched with validUntil in miliseconds from quotes instead 30000
```

Comment:
Why can't it be done yet?

Reply:
Because I've requested that the new value comes in the response, at this moment they are just responding the date